datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -727,6 +727,8 @@ public SdxCluster save(SdxCluster sdxCluster) {
     }
 
     public void renewCertificate(SdxCluster sdxCluster, String userCrn) {
-        stackV4Endpoint.renewCertificate(0L, sdxCluster.getClusterName(), userCrn);
+        ThreadBasedUserCrnProvider.doAsInternalActor(() -> {
```

Comment:
should we cover this with a unit test ? I guess this is a bugfix so would be great to see some unit change for this.

Reply:
unfortunately, it seems to be hard to test this. The doAsInternalActor modifies a thread-local variable and later the ApiClientRequestFilter reads this in filterchain, so in unit test there is no chance to test this, maybe with some spring integration test.